src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionRequestHandler.java

```diff
@@ -411,6 +412,37 @@ private PixelBuffer getPixelBuffer(Pixels pixels)
         return null;
     }
 
+    /**
+     * Mirror an image over the X, Y or both axes.
+     * @param src source image buffer
+     * @param sizeX size of <code>src</code> in X (number of columns)
+     * @param sizeY size of <code>src</code> in Y (number of rows)
+     * @param mirrorX whether or not to mirror the image over the X axis
+     * @param mirrorY whether or not to mirror the image over the Y axis
+     * @return Newly allocated buffer with mirroring applied or <code>src</code>
+     * if no mirroring has been requested.
+     */
+    public static int[] mirror(
+            int[] src, int sizeX, int sizeY, boolean mirrorX, boolean mirrorY) {
+        if (!mirrorX && !mirrorY) {
+            return src;
+        }
+
+        int[] dest = new int[src.length];
+        int srcIndex, destIndex;
+        int xOffset = mirrorX? sizeX : 1;
+        int yOffset = mirrorY? sizeY : 1;
+        for (int x = 0; x < sizeX; x++) {
+            for (int y = 0; y < sizeY; y++) {
+                srcIndex = (y * sizeX) + x;
+                destIndex = Math.abs(((yOffset - y - 1) * sizeX))
+                        + Math.abs((xOffset - x - 1));
```

Comment:
Might be easier to understand if you change xOffset and yOffset to be size - 1 if you mirror and 0 otherwise. Then you wouldn't need these  - 1s here.